elliptic-curve/src/lib.rs

```diff
@@ -115,10 +122,15 @@ pub trait FromDigest<C: Curve> {
         D: Digest<OutputSize = C::FieldSize>;
 }
 
-/// Associate an object identifier (OID) with a curve
-#[cfg(feature = "oid")]
-#[cfg_attr(docsrs, doc(cfg(feature = "oid")))]
-pub trait Identifier: Curve {
+/// Associate an [`ObjectIdentifier`][`pkcs8::ObjectIdentifier`] (OID) with an
+/// elliptic curve algorithm implementation.
+///
+/// This is used as as the `parameters` of an `AlgorithmIdentifier` as
+/// described in RFC 5280 Section 4.1.1.2:
+/// <https://tools.ietf.org/html/rfc5280#section-4.1.1.2>
+#[cfg(feature = "pkcs8")]
+#[cfg_attr(docsrs, doc(cfg(feature = "pkcs8")))]
+pub trait AlgorithmParameters: Curve {
```

Comment:
Not sure about this name. I chose it to distinguish it from `pkcs8::AlgorithmIdentifier`, and because this OID is ultimately stored in the `parameters` field of that struct, but while that appropriately names the field in the ASN.1 schema, the phrase `AlgorithmParameters` doesn't exactly scream "this is the trait you use to associate an OID with a curve implementation" to me.